crates/compiler/mono/src/ir.rs

```diff
@@ -9983,3 +10063,48 @@ where
 
     answer
 }
+
+enum Usage {
+    Used,
+    Unused,
+}
+
+pub struct UsageTrackingMap<K, V>
+where
+    K: std::cmp::Eq + std::hash::Hash,
+{
+    map: MutMap<K, (V, Usage)>,
+}
+
+impl<K, V> Default for UsageTrackingMap<K, V>
+where
+    K: std::cmp::Eq + std::hash::Hash,
+{
+    fn default() -> Self {
+        Self {
+            map: MutMap::default(),
+        }
+    }
+}
```

Comment:
if you remove the abound on the struct definition, you should be able to just derive this instance

Reply:
I can derive it, but it wants `Symbol` to implement `Default` as well. So I left it as is.